src/NumericalAlgorithms/Spectral/SwshTransformJob.cpp

```diff
@@ -154,17 +167,36 @@ SpinWeighted<ComplexModalVector, Spin> swsh_transform(
   return result_vector;
 }
 
+template <int Spin>
```

Comment:
Instead of this overload use a default template parameter for the Representation.

Reply:
This one is troublesome for template parameter ordering. The point here is that the `Spin` parameter should always be inferred from the spin-weight of the argument, and that template defaults must follow non-defaulted template parameters. If the spin was first, however, it would try to match to the first specified template parameter in calling code. So, as far as I can tell, the template system does not permit from a single definition both the possibility of specifying only the representation and inferring the spin or having no specified parameters and defaulting the representation while inferring the spin.

Do you have a way around this template parameter ordering trouble?

Reply:
I think you can actually have inferred template parameters coming after the defaulted template parameters.
By I think is, I wrote a test code doing this, which is compiling without complaints and is working. Furthermore I think there is no standard actually prohibiting this.

This is my test code:
```
#include <iostream>

template <int Spin>
class SpinWeighted {};

enum class ComplexRepresentation { Interleaved, RealsThenImags };

template <ComplexRepresentation Representation=ComplexRepresentation::Interleaved,
          int Spin>
void inverse_swsh_transform(
    SpinWeighted<Spin> input) noexcept{
    std::cout << "Spin: " << Spin << std::endl << "Representation: ";
    if(Representation==ComplexRepresentation::Interleaved)
        std::cout << "Interleaved";
    else
        std::cout << "RealsThenImags";
    std::cout << std::endl;
}
main() {
  SpinWeighted<1> in;
  inverse_swsh_transform<ComplexRepresentation::RealsThenImags>(in);
  inverse_swsh_transform(in);
}
```

Reply:
Aha, excellent. I must have misinterpreted a compiler error at some point when putting this together. Because the change effectively reverts one of the commits, I've squashed the change into that commit, and relabeled it with a `fixup` prefix to ultimately collapse into the other `SwshTransform` changes.